com.unity.multiplayer.mlapi/Tests/Editor/ProfilerTests.cs

```diff
@@ -0,0 +1,192 @@
+        {
+            PerformanceDataManager.Increment(ProfilerConstants.TransportTestData);
+        }
+    }
+
+    public class TestHasProfilable : IHasProfilableTransport
+    {
+        internal static class ProfilerConstants
+        {
+            public const string NetworkTestData = nameof(NetworkTestData);
+        }
+
+        private TestTransport m_Transport;
+
+        public ITransportProfilerData GetTransport()
+        {
+            return m_Transport;
+        }
+
+        public void Initialize(bool useNullTransport)
+        {
+            m_Transport = useNullTransport ? null : new TestTransport();
+            ProfilerNotifier.Initialize(this);
+        }
+
+        public static void ProfilerBeginTick()
+        {
+            ProfilerNotifier.ProfilerBeginTick();
+        }
+
+        public static void NotifyProfilerListeners()
+        {
+            ProfilerNotifier.NotifyProfilerListeners();
+        }
+
+        public void Send()
+        {
+            ProfilerNotifier.Increment(ProfilerConstants.NetworkTestData);
+            m_Transport?.Send("testMessage");
+        }
+    }
+
+    public class NoTickDataException : Exception
+    {
+    }
+
+    public class ProfilerTests
+    {
+        private static void BreakDownTestProfiler(bool useNullTransport)
+        {
+            if (useNullTransport)
+            {
+                ProfilerNotifier.OnPerformanceDataEvent -= TestProfilerOnPerformanceDataEventNoTransport;
+            }
+            else
+            {
+                ProfilerNotifier.OnPerformanceDataEvent -= TestProfilerOnPerformanceDataEventNormal;
+            }
+
+            ProfilerNotifier.OnNoTickDataEvent -= TestProfilerNotifierOnOnNoTickDataEvent;
+        }
+
+        private static TestHasProfilable SetupTestProfiler(bool useNullTransport)
+        {
+            ProfilerNotifier.OnNoTickDataEvent += TestProfilerNotifierOnOnNoTickDataEvent;
+            if (useNullTransport)
+            {
+                ProfilerNotifier.OnPerformanceDataEvent += TestProfilerOnPerformanceDataEventNoTransport;
+            }
+            else
+            {
+                ProfilerNotifier.OnPerformanceDataEvent += TestProfilerOnPerformanceDataEventNormal;
+            }
+
+            EditorApplication.UnlockReloadAssemblies();
+            var testProfiler = new TestHasProfilable();
+            testProfiler.Initialize(useNullTransport);
+            return testProfiler;
+        }
+
+        [Test]
+        public void TestNormalRegisterAndNotifyFlowNull()
+        {
+            const bool useNullTransport = true;
+            TestHasProfilable testProfiler = SetupTestProfiler(useNullTransport);
+
+            TestHasProfilable.ProfilerBeginTick();
+            testProfiler.Send();
+            TestHasProfilable.NotifyProfilerListeners();
+
+            BreakDownTestProfiler(useNullTransport);
+        }
+
+        [Test]
+        public void TestNormalRegisterAndNotifyFlow()
```

Comment:
About this test, and `TestNormalRegisterAndNotifyFlowNull`, there's no asserts in the test and the method name doesn't describe how it could fail - if this test were to fail on me due to some work I was doing, it would be hard for me to decipher what the intent of the test was. If at all possible I'd recommend adding explicit asserts for the fail cases you're testing